### ISSUE TITLE
Wifi sometimes stop working - not hardware installed

### ISSUE THREAD

**Issue by alexmenn** [daliansky/XiaoMi-Pro#95]
Hello all! I'm working with internal dw1560 in my notebook pro and everything is OK, I'm using original antennas and speed is great. But sometimes, after sleep, the laptop says that there is not hardware installed, and I have to reboot to fix it.

Sometimes (less times), when I reboot, after boot, the systems says too there is not wifi hardware installed, and I have to reboot again. Any clue?

**Comment by influenist** [comment]
@alexmenn you have this issue since you updated your EFI recently?

Anyone with internal BMC having this same issue?

**Comment by alexmenn** [comment]
No, I have had this problem since HS :)

**Comment by stevezhengshiqi** [comment]
@alexmenn Hi, since it’s a problem related to third party hardware, you can send trouble shooting to tonymacx86.com to get professional help.

**Comment by alexmenn** [comment]
@stevezhengshiqi Ok, I will ask. Can you give me link to buy your BCM943602CS to check it?

**Comment by Menchen** [comment]
@alexmenn I think that is wrong adaptor,... you need key m or b, the link you provided is key a/e so it won't fit, also about the antenna I'm sure you can connect wifi one to original port and Bluetooth one under trackpad, but, you may want keep original intel wifi card as the apple one don't work well on linux and windows(need driver), as discussed on steve repo, go check them.

**Comment by alexmenn** [comment]
You're right. Key m or b, ok, but what kind of connection uses the chip BCM943602CS? I cant find any on ebay.

**Comment by Menchen** [comment]
@alexmenn I think it's apple own one connection,12+6 pin, and that port to key m is pretty rare... But I found one on aliexpress for you.

Only adaptor .Bluetooth should work if you solder usb bus.
https://m.aliexpress.com/item/32949190082.html?pid=808_0000_0101&spm=a2g0n.search-amp.list.32949190082&aff_trace_key=&aff_platform=msite&m_page_id=1488amp-N9L0BcBVwnB6p_6y_KarrQ1541246198851

**Comment by alexmenn** [comment]
Hmm seems to be perfect. When I cut it, will it fit in the m2 slot?

**Comment by Menchen** [comment]
@alexmenn I don't own that adaptor, but it should fit without problem, as the m.2 size that we have is pretty standard.(designed to fit a ssd, with is longer than wifi card.)

**Comment by Menchen** [comment]
@alexmenn you also need antenna.... 3 antenna or two wire to connect to original one (I think you already have it.)plus a antenna for Bluetooth. Also usb wire,normal wire that you can find on local shop? Not sure about USB wire is equal than a normal wire....  and it's seem that adaptor have a usb socket maybe you should search more about it..

**Comment by alexmenn** [comment]
I have few wires to connect original antenna to chip right now. I just need one more antenna for BT, but I think I wont use BT with this chip. I just need a stable wifi which works after sleep :)

**Comment by Menchen** [comment]
@alexmenn well it's up to you, but Bluetooth is powered.. so wasted power. (You also have paid for a card that have native Bluetooth,  so no more harassment kext)

**Comment by alexmenn** [comment]
Hmm ok, bluetooth will be nice, but how can I "connect it"? Im sure I will have to solder some wires...
